Reservoir and wetlands updates

PeterBurek opened the pull request:
**Description**
Optional changes to Lisflood (if added options set to 0, the model runs like v4.3.1)
Full description in:
https://confluence.ecmwf.int/display/SEEDFD/Lisflood+code+change+for+wetlands+and+reservoirs

Added wetlands: for bigger seasonal wetlands e.g Sudd(Nile), Inner Delta Niger and additional module is added, which takes care of seasonal flooding and increased evaporation from flooded area. Wetlands are put in as a single point in an Excel file with an additional 366 day table of variable flooding area

Changed evapowater (optional). Water from lakes, wetlands, reservoir is not taken out via fraction of water x EWRef, but directly in the lakes, wetland, reservoir routine by calculating

Evaporation = waterbody area x EWRef per routing step. This amount is then taken from the storage of the waterbody

Gridcells of lakemask upstream of each res/wetland/lake grid point are taken out for the calculation of additional evaporation from open water to avoid double counting

Lakes/reservoir: Added read in from an Excel file for additional waterbodies. In addition to the maps and tables in Lakes_Reservoirs_ID_tables, waterbodies can be read in as single points from the Excel or changes to attributes of existing waterbodies can be done. The Excel attributes will overwrite the maps/table attribute if different than -1.

 **Additional Libraries**
- openpyxl: for using Excel sheet
For groundwater (later):
- flopy
- xmipy
- rasterio (will be replaced with excisting library - e.g rioxarray)

**New modules**
wetlands.py: wetland module – similar to lakes.py but for periodical wetlands
transient.py: module for coupling lisflood<> modflow6
modflow6.py: modflow6  coupler using libraries flopy and xmipy

**Tests**

63 test pasted
New tests in: test_latlon_lakeres.py

madiej6 committed:
Update index.md

fix some typos

doc78 committed:
Merge pull request #147 from madiej6/patch-1

Update index.md

doc78 committed:
Small update to docs for preparation of setting files

doc78 committed:
Small update in docs on how to use multiple nc input files for meteo forcings

PeterBurek committed:
adding components for Lisflood-Modflow coupling
modflow6: triggering modflow6.py
transient: projecting necessaery data from Lisflood lat/lon to rectangular modflow6

PeterBurek committed:
changing reservoir.py to read excel with addition reservoirs
adding excel with additional reservoir data

PeterBurek committed:
added reading of excel witk lakes, reservoirs, wetland
added reservoir release curve reading and using
added reservoir type 2 (as before) , 3 (with release)
toDo: wetland

PeterBurek committed:
added: wetland.py with variable area per day
changed: evaporation from lakes and wetlands are calculated and is reducing storage_channel
changed: evapowater.py: lake and wetlandland are excluded from open water evaporation
Changed: Excel file with additional reservoirs include daily change of wetland area

PeterBurek committed:
Changes to lakes and wetlands, groundwater.
Evapowater is (optional) changed -> direct caluclation of lake/res/wetland evapo from area
This affects also global modules like add.py, output.py (and others)

PeterBurek committed:
change in wetlands
change in tests - add an additional testr
documentation in: https://confluence.ecmwf.int/display/SEEDFD/Lisflood+code+change+for+wetlands+and+reservoirs

nicola-martin commented on docs/3_step3_preparing-setting-file/lisfloodSettings_reference.xml:
Has this file been deleted deliberately? Is there a new settings reference file that has the extra options?

nicola-martin replied:
Ok, I see that it is in src/ now. Are there any updates needed in this reference file?

PeterBurek replied:
No, I still have to update this. It should work anyway without the new feature
An updated xml in tests/data/LF_lat_lon_UseCase/run_lat_lon2.xml
which addition data in tests/data/LF_lat_lon_UseCase/tables  (rarea_687.txt and lakes_reservoirs.xlsx)

PeterBurek committed:
-added table/wetlandavinflow_463.txt
- changed requirements - still trouble with GDAL therefore as last
   error install GDAL -> downscale setuptools
    pip install setuptools==57.5.0
    pip install GDAL==3.2.1

PeterBurek committed:
Corrected evapowater.py to run if InitLisflood = 1

nicola-martin commented:
I have tested out the branch in the longflow suite on the ECMWF systems.
The existing and new python tests are all succeeding.
Running with the new features switched off reproduces the results of the seed-fd branch.
The suite runs through successfully with the new reservoir and wetlands options switched on.

There are still updates to come and the ground water changes are not yet ready for testing, but I'm happy to merge in the latest changes into the seed-fd branch for use in the prototype runs.
